app/models/event.rb

```diff
@@ -0,0 +1,57 @@
+# == Schema Information
+#
+# Table name: events
+#
+#  id              :bigint           not null, primary key
+#  data            :jsonb
+#  handleable_type :string
+#  name            :string
+#  type            :string
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#  handleable_id   :bigint
+#
+# Indexes
+#
+#  index_events_on_handleable_type_and_handleable_id  (handleable_type,handleable_id)
+#
+
+class Event < ApplicationRecord
+  belongs_to :handleable, polymorphic: true, required: false
+  validates :name, :data, presence: true
+
+  EVENTS = %w[pull_request].freeze
+  private_constant :EVENTS
+
+  def resolve
+    return handle if handleable?
+
+    save!
+  end
+
+  private
+
+  def handle
+    Event.create!(build_attributes)
```

Comment:
Are we duplicating the `Event` here? 🤔

Reply:
If it is handleable it will create a `Event` with, for example, `PullRequest`, otherwise it will create a `Event` with just the name and payload.
Now we have one `PullRequest` which has multiple `Events`.